corehq/apps/reports/tasks.py

```diff
@@ -337,11 +338,33 @@ def extract_form_info(form, properties=None, case_ids=case_ids):
 
         return form_info
 
-    key = [domain, app_id, xmlns]
-    form_ids = {f['id'] for f in XFormInstance.get_db().view("attachments/attachments",
-                                                             start_key=key + [startdate],
-                                                             end_key=key + [enddate, {}],
-                                                             reduce=False)}
+    if not export_is_legacy:
+        query = (FormES()
+                 .domain(domain)
+                 .app(app_id)
+                 .xmlns(xmlns)
+                 .submitted(gte=parse(startdate), lte=parse(enddate))
+                 .remove_default_filter("has_user"))
+        form_ids = set()
+        for form in query.scroll():
+            try:
+                for attachment in form['_attachments'].values():
```

Comment:
we could also write a helper function to take care of the dirty work. Something like:
```py
for form in query.scroll():
    # get consolidated dict of _attachments and external_documents
    blobs = get_es_blob_meta(form)
    for attachment in blobs.values():
        ...
```
Not sure if we would also want/need to write a similar function for querying ES docs based on attachment metadata.